
<launch>
  <node pkg="project1" type="talkerGoals" name="talkerGoals" output="screen">
    <param name="X_GOAL" value="8.0"/>
    <param name="Y_GOAL" value="6.0"/>
    <param name="T_GOAL" value="5.0"/>
  </node>

  <param name="NUM_ROBOTS" value="2"/>

<!-- Leader -->
<node pkg="project1" type="moveRobot" name="moveRobot0" output="screen">
    <param name="ID_ROBOT" value="0"/>
    <param name="LEADER" value="-1"/>

    <!-- Missions -->
    <param name="ALGOR" value="1"/>   
    <param name="ROBOT_ROL" value="0"/>   <!-- Leader -->

    <!-- Initial pose -->
    <param name="X_START" value="4.0"/>
    <param name="Y_START" value="-10.0"/>

    <!-- Behavior parameters -->
    <param name="CRIT_DIST" value="1.0"/>        <!-- 0.8–1.2 -->
    <param name="D_OBJ" value="0.4"/>            <!-- 0.2–0.6 -->
    <param name="V_MAX_DES" value="0.8"/>        <!-- 0.8–1.2 -->
    <param name="V_MAX_ROT" value="0.9"/>        <!-- 0.8–1.1 -->

    <!-- Proportional controller -->
    <param name="K_ROT_MIN" value="0.10"/>       <!-- 0.05–0.15 -->
    <param name="K_ROT_MAX" value="0.25"/>       <!-- 0.2–0.3 -->

    <param name="ORI_ERROR" value="0.4"/>        <!-- 0.3–0.6 -->
    <param name="T_AVOID_OBS" value="2.0"/>      <!-- 1.0–3.0 -->

    <param name="T_WAIT" value="150"/>           <!-- 100–250 ms -->
  </node>


 <!-- Follower -->
  <node pkg="project1" type="moveRobot" name="moveRobot1" output="screen">
    <param name="ID_ROBOT" value="1"/>
    <param name="LEADER" value="0"/>

    <!-- Missions -->
    <param name="ALGOR" value="1"/>   
    <param name="ROBOT_ROL" value="1"/>   <!-- Follower -->
    <param name="ID_LEADER" value="0.0"/>

    <!-- Initial pose -->
    <param name="X_START" value="7.0"/>
    <param name="Y_START" value="-7.0"/>

    <!-- Follower-specific -->
    <param name="DIST_LEADER" value="2.0"/>      <!-- 1.5–2.5 -->

    <!-- Behavior parameters -->
    <param name="CRIT_DIST" value="1.0"/>
    <param name="D_OBJ" value="0.4"/>
    <param name="V_MAX_DES" value="1.0"/>
    <param name="V_MAX_ROT" value="0.9"/>

    <!-- Proportional controller -->
    <param name="K_ROT_MIN" value="0.10"/>
    <param name="K_ROT_MAX" value="0.25"/>

    <param name="ORI_ERROR" value="0.4"/>
    <param name="T_AVOID_OBS" value="2.0"/>

    <param name="T_WAIT" value="150"/>
  </node>


  <!-- node pkg="project1" type="moveRobot" name="moveRobot2" output="screen">
    <param name="ID_ROBOT" value="2"/>
	<param name="LEADER" value="1"/>
    <param name="ALGOR" value="1"/>
	<param name="X_START" value="0.0"/>
	<param name="Y_START" value="-7.0"/>
  </node -->

  <!-- node pkg="project1" type="moveRobot" name="moveRobot3" output="screen">
    <param name="ID_ROBOT" value="3"/>
	<param name="LEADER" value="0"/>
    <param name="ALGOR" value="1"/>
	<param name="X_START" value="-5.0"/>
	<param name="Y_START" value="7.0"/>
  </node -->

  <!-- Generate world with goal area and its robots -->
  <node pkg="project1" type="world_generator" name="world_generator" output="screen"/>

  <!-- Launch Stage with the generated world -->
  <node name="Stage" pkg="stage_ros" type="stageros" args="$(find project1)/worlds/mundo.world"/>
</launch>
